backend/models/role.js

```diff
@@ -16,95 +16,72 @@
-
-		return Role.findByRoleID(roleId).then(roleFound => {
-			if(roleFound) {
-				roleFound = roleFound.toObject();
-				return { role: roleFound.role, db: roleFound.db};
-			} else {
-				const roleName = C.DEFAULT_MEMBER_ROLE;
-				const createRoleCmd = {
-					"createRole": roleName,
-					"privileges":[{
-						"resource":{
-							"db": account,
-							"collection": "settings"
-						},
-						"actions": ["find"]}
-					],
-					"roles": []
-				};
-
-				return ModelFactory.dbManager.runCommand(account, createRoleCmd).then(()=> {
-					return {role: roleName, db: account};
-				});
-			}
-		});
 
-	};
-
-	schema.statics.dropTeamSpaceRole = function (account) {
-		const dropRoleCmd = {
-			"dropRole" : C.DEFAULT_MEMBER_ROLE
-		};
+const C = require("../constants");
+const DB = require("../handler/db");
 
-		return this.findByRoleID(`${account}.${C.DEFAULT_MEMBER_ROLE}`).then(role => {
-			if(!role) {
-				return Promise.resolve();
-			} else {
-				return ModelFactory.dbManager.runCommand(account, dropRoleCmd);
-			}
-		});
+const Role = {};
 
-	};
+Role.createTeamSpaceRole = async function (account) {
+	const roleId = `${account}.${C.DEFAULT_MEMBER_ROLE}`;
 
-	schema.statics.grantTeamSpaceRoleToUser = function (username, account) {
+	const roleFound = await Role.findByRoleID(roleId);
 
-		const grantRoleCmd = {
-			grantRolesToUser: username,
-			roles: [{role: C.DEFAULT_MEMBER_ROLE, db: account}]
-		};
+	if(roleFound) {
+		return { role: roleFound.role, db: roleFound.db};
+	}
 
-		return ModelFactory.dbManager.runCommand("admin", grantRoleCmd);
+	const roleName = C.DEFAULT_MEMBER_ROLE;
+	const createRoleCmd = {
+		"createRole": roleName,
+		"privileges":[{
+			"resource":{
+				"db": account,
+				"collection": "settings"
+			},
+			"actions": ["find"]}
+		],
+		"roles": []
 	};
 
-	schema.statics.findByRoleID = function(id) {
-		return this.findOne({ account: "admin"}, { _id: id});
+	await DB.runCommand(account, createRoleCmd);
+	return {role: roleName, db: account};
+};
+
+Role.dropTeamSpaceRole = async function (account) {
+	const dropRoleCmd = {
+		"dropRole" : C.DEFAULT_MEMBER_ROLE
 	};
 
-	schema.statics.revokeTeamSpaceRoleFromUser = function(username, account) {
+	const role = await this.findByRoleID(`${account}.${C.DEFAULT_MEMBER_ROLE}`);
 
-		const cmd = {
-			revokeRolesFromUser: username,
-			roles: [{role: C.DEFAULT_MEMBER_ROLE, db: account}]
-		};
+	if(!role) {
+		return;
+	}
 
-		return ModelFactory.dbManager.runCommand("admin", cmd);
+	return await DB.runCommand(account, dropRoleCmd);
+};
+
+Role.grantTeamSpaceRoleToUser = async function (username, account) {
+	const grantRoleCmd = {
+		grantRolesToUser: username,
+		roles: [{role: C.DEFAULT_MEMBER_ROLE, db: account}]
 	};
 
-	const Role = ModelFactory.createClass(
-		"Role",
-		schema,
-		() => {
-			return "system.roles";
-		}
-	);
+	return await DB.runCommand("admin", grantRoleCmd);
+};
+
+Role.findByRoleID = async function(id) {
```

Comment:
This should be a private function. It's only used in this file. (and in context of 3D Repo it'll never be used anywhere else)